nx/lib/nx/serving.ex

```diff
@@ -310,12 +310,17 @@ defmodule Nx.Serving do
       client_postprocessing: postprocessing
     } = serving
 
-    {:ok, state} = handle_init(module, :inline, arg)
-    {%{size: size} = batch, info} = handle_preprocessing(preprocessing, input)
-    {:execute, function, _} = handle_batch(module, batch, state)
-    {output, metadata} = handle_executed(module, function.())
-    output = Nx.Defn.Composite.traverse(output, &Nx.slice_along_axis(&1, 0, size, axis: @axis))
-    handle_postprocessing(postprocessing, output, metadata, info)
+    meta = %{arg: arg, input: input, module: module}
+
+    :telemetry.span([:nx, :serving, :run], meta, fn ->
```

Comment:
Instead of wrapping the whole thing, what if we wrap only `handle_executed` with an event of `:execute`? It is better as it applies to both batched_run and run.

Reply:
Sounds good to me!

Reply:
Did you mean `handle_batch`? The `handle_executed` function only validates that the result is a two-element tuple.

Reply:
@sorentwo yes, I meant the call to that exact function!